packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -2232,53 +2159,73 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
+  /// Set the current [selection] to contain the entire text value.
+  ///
+  /// {@macro flutter.rendering.RenderEditable.cause}
+  void selectAll(SelectionChangedCause cause) {
+    _setSelection(
+      selection!.copyWith(
+        baseOffset: 0,
+        extentOffset: textSelectionDelegate.textEditingValue.text.length,
+      ),
+      cause,
+    );
+  }
+
+  /// Copy current [selection] to Clipboard.
+  void copySelection() {
     final TextSelection selection = textSelectionDelegate.textEditingValue.selection;
     final String text = textSelectionDelegate.textEditingValue.text;
     assert(selection != null);
-    assert(_shortcutKeys.contains(key), 'shortcut key $key not recognized.');
-    if (key == LogicalKeyboardKey.keyC) {
-      if (!selection.isCollapsed) {
-        Clipboard.setData(ClipboardData(text: selection.textInside(text)));
-      }
+    if (!selection.isCollapsed) {
+      Clipboard.setData(ClipboardData(text: selection.textInside(text)));
+    }
+  }
+
+  /// Cut current [selection] to Clipboard.
+  ///
+  /// {@macro flutter.rendering.RenderEditable.cause}
+  void cutSelection(SelectionChangedCause cause) {
+    if (_readOnly) {
       return;
     }
-    TextEditingValue? value;
-    if (key == LogicalKeyboardKey.keyX && !_readOnly) {
-      if (!selection.isCollapsed) {
-        Clipboard.setData(ClipboardData(text: selection.textInside(text)));
-        value = TextEditingValue(
+    final TextSelection selection = textSelectionDelegate.textEditingValue.selection;
+    final String text = textSelectionDelegate.textEditingValue.text;
+    assert(selection != null);
+    if (!selection.isCollapsed) {
+      Clipboard.setData(ClipboardData(text: selection.textInside(text)));
+      _setTextEditingValue(
+        TextEditingValue(
           text: selection.textBefore(text) + selection.textAfter(text),
           selection: TextSelection.collapsed(offset: math.min(selection.start, selection.end)),
-        );
-      }
-    } else if (key == LogicalKeyboardKey.keyV && !_readOnly) {
-      // Snapshot the input before using `await`.
-      // See https://github.com/flutter/flutter/issues/11427
-      final ClipboardData? data = await Clipboard.getData(Clipboard.kTextPlain);
-      if (data != null && selection.isValid) {
-        value = TextEditingValue(
-          text: selection.textBefore(text) + data.text! + selection.textAfter(text),
-          selection: TextSelection.collapsed(
-            offset: math.min(selection.start, selection.end) + data.text!.length,
-          ),
-        );
-      }
-    } else if (key == LogicalKeyboardKey.keyA) {
-      value = TextEditingValue(
-        text: text,
-        selection: selection.copyWith(
-          baseOffset: 0,
-          extentOffset: textSelectionDelegate.textEditingValue.text.length,
         ),
+        cause,
       );
     }
-    if (value != null) {
+  }
+
+  /// Paste text from Clipboard, if selection exists, then selection will be replaced.
```

Comment:
```diff
- /// Paste text from Clipboard, if selection exists, then selection will be replaced.
+ /// Paste text from [Clipboard].
+ ///
+ /// If there is currently a selection, it will be replaced.
```